src/test/java/io/confluent/copycat/hdfs/FSWALTest.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.copycat.hdfs;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.kafka.copycat.errors.CopycatException;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class FSWALTest extends HdfsSinkConnectorTestBase {
+
+  @Test
+  public void testWALMultiClient() throws Exception {
+    fs.delete(new Path(FileUtils.directoryName(url, topicsDir, TOPIC_PARTITION)), true);
+
+    Class<? extends Storage> storageClass = (Class<? extends Storage>) Class.forName("io.confluent.copycat.hdfs.HdfsStorage");
+    Storage storage1 = StorageFactory.createStorage(storageClass, conf, url);
+    Storage storage2 = StorageFactory.createStorage(storageClass, conf, url);
+
+    final WAL wal1 = WALFactory.createWAL(topicsDir, TOPIC_PARTITION, storage1);
+    final WAL wal2 = WALFactory.createWAL(topicsDir, TOPIC_PARTITION, storage2);
+
+    final String tempfile = FileUtils.tempFileName(url, topicsDir, TOPIC_PARTITION);
+    final String commitedFile = FileUtils.committedFileName(url, topicsDir, TOPIC_PARTITION, 0, 10);
+
+    fs.createNewFile(new Path(tempfile));
+
+    Thread thread = new Thread(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          wal1.acquireLease();
```

Comment:
Since you're just waiting for this thread to exit anyway, there's no need for it to be a separate thread. Just invoke the methods inline in the test.